include/seqan3/utility/type_list/traits.hpp

```diff
@@ -114,6 +114,33 @@ auto reverse(type_list<head_t, pack_t...>)
     return concat(reverse(type_list<pack_t...>{}), type_list<head_t>{});
 }
 
+//!\brief Constructs the set difference `list1 \ list2` [recursion anchor]
+template <typename ...elements1_t>
+constexpr seqan3::type_list<elements1_t...> type_list_difference(seqan3::type_list<elements1_t...>, seqan3::type_list<>)
+{ return {}; }
+
+//!\brief Constructs the set difference `list1 \ list2` [recursion]
+template <typename ...elements1_t, typename head2_t, typename ...elements2_t>
+constexpr auto type_list_difference(seqan3::type_list<elements1_t...>, seqan3::type_list<head2_t, elements2_t...>)
+{
+    constexpr auto pos = seqan3::pack_traits::find<head2_t, elements1_t...>;
+    if constexpr (pos >= 0)
+    {
+        using split_list_t = seqan3::pack_traits::split_after<pos, elements1_t...>;
+
+        using split_list1_t = typename split_list_t::first_type;
+        using split_list2_t = decltype(drop_front(typename split_list_t::second_type{}));
+        using filtered_elements_t = decltype(concat(split_list1_t{}, split_list2_t{}));
+        return type_list_difference(filtered_elements_t{}, seqan3::type_list<head2_t, elements2_t...>{});
+    }
+    else
+    {
+        // head2_t not contained in elements1_t
+        using filtered_elements_t = seqan3::type_list<elements1_t...>;
+        return type_list_difference(filtered_elements_t{}, seqan3::type_list<elements2_t...>{});
+    }
+}
```

Comment:
I think this has the problem that it will only remove the first occurrence?

I also added a comment in the tests, but I think that `{float, double, float, double} \ {float}` will be `{double, float, double}`, didn't test it though.

Reply:
* `{float, double, float, double} \ {float}`
* => goes into if
* => `({} ∪ {double, float, double}) \ {float}`
* => `{double, float, double} \ {float}`
* => goes into if
* => `({double} ∪ {double}) \ {float}`
* => `{double, double} \ {float}`
* => goes into else
* => `{double, double} \ {}`
* => goes into anchor
* => `{double, double}`

Reply:
This tested here:

https://github.com/seqan/seqan3/pull/2614/files#r632463176

Reply:
Hah yes.
I forgot to do the actual recursion :D